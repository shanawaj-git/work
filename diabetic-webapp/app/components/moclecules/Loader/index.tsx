import React from 'react';
import Box from '@mui/material/Box';
import PropTypes, { InferProps } from 'prop-types';

import CircularProgress from '@mui/material/CircularProgress';

const Loader = ({ size = 95, thickness = 2 }: LoaderProps) => (
  <Box sx={{ position: 'relative', width: size, height: size }}>
    <CircularProgress
      variant="determinate"
      sx={{
        color: '#c5c5c5',
      }}
      size={size}
      thickness={thickness}
      value={100}
    />
    <svg style={{ position: 'absolute' }}>
      <defs>
        <linearGradient id="my_gradient">
          <stop offset="0%" stopColor="#00664f" />
          <stop offset="100%" stopColor="#c5c5c5" />
        </linearGradient>
      </defs>
    </svg>
    <CircularProgress
      variant="indeterminate"
      disableShrink
      sx={{
        'svg circle': { stroke: 'url(#my_gradient)' },
        position: 'absolute',
        left: 0,
      }}
      size={size}
      thickness={thickness}
    />
  </Box>
);

const LoaderPropTypes = {
  size: PropTypes.number,
  thickness: PropTypes.number,
};

type LoaderProps = InferProps<typeof LoaderPropTypes>;

Loader.propTypes = LoaderPropTypes;

export default Loader;
