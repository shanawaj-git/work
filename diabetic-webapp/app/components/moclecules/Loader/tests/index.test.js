import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Loader from '../index';

const renderer = new ShallowRenderer();

describe('<Loader />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<Loader />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
