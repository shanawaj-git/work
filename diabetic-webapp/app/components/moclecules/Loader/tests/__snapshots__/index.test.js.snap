// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render and match the snapshot 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "height": 95,
      "position": "relative",
      "width": 95,
    }
  }
>
  <ForwardRef(CircularProgress)
    size={95}
    sx={
      Object {
        "color": "#c5c5c5",
      }
    }
    thickness={2}
    value={100}
    variant="determinate"
  />
  <svg
    style={
      Object {
        "position": "absolute",
      }
    }
  >
    <defs>
      <linearGradient
        id="my_gradient"
      >
        <stop
          offset="0%"
          stopColor="#00664f"
        />
        <stop
          offset="100%"
          stopColor="#c5c5c5"
        />
      </linearGradient>
    </defs>
  </svg>
  <ForwardRef(CircularProgress)
    disableShrink={true}
    size={95}
    sx={
      Object {
        "left": 0,
        "position": "absolute",
        "svg circle": Object {
          "stroke": "url(#my_gradient)",
        },
      }
    }
    thickness={2}
    variant="indeterminate"
  />
</ForwardRef(Box)>
`;
